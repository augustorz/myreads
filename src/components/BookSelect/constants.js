export const SHELVES = [
  { value: 'currentlyReading', text: 'Currently Reading' },
  { value: 'wantToRead', text: 'Want to Read' },
  { value: 'read', text: 'Read' },
  { value: '', text: 'None' },
];

export default {
  SHELVES,
};
