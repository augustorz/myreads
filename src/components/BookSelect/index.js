import React from 'react';
import PropTypes from 'prop-types';
import { SHELVES } from './constants';

const BookSelect = ({ book, onChangeBookShelf }) => {
  const onChange = ({ target: { value: newShelf } }) => {
    onChangeBookShelf({ book, newShelf });
  };

  return (
    <div className="book-shelf-changer">
      <select
        value={book.shelf}
        onChange={onChange}
      >
        <option value="move" disabled>
          Move to...
        </option>
        {
          SHELVES.map(({ value, text }) => (
            <option key={value} value={value}>
              {text}
            </option>
          ))
        }
      </select>
    </div>
  );
};

BookSelect.propTypes = {
  book: PropTypes.instanceOf(Object).isRequired,
  onChangeBookShelf: PropTypes.func.isRequired,
};

export default BookSelect;
