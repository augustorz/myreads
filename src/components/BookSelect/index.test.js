import React from 'react';
import { shallow } from 'enzyme';
import createBooks from '../../mocks/books';

import BookSelect from '.';

describe('BookSelect', () => {
  let wrapper;

  beforeEach(() => {
    const book = createBooks({ shelf: 'wantToRead' });

    wrapper = shallow(
      <BookSelect
        book={book}
        onChangeBookShelf={jest.fn()}
      />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('onChange', () => {
    beforeEach(() => {
      
    });
  });
});
