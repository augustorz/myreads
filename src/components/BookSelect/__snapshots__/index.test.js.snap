// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookSelect should match snapshot 1`] = `"<div class=\\"book-shelf-changer\\"><select><option selected=\\"\\" value=\\"move\\" disabled=\\"\\">Move to...</option><option value=\\"currentlyReading\\">Currently Reading</option><option value=\\"wantToRead\\">Want to Read</option><option value=\\"read\\">Read</option><option value=\\"\\">None</option></select></div>"`;
